datalad_next/constraints/dataset.py

```diff
@@ -50,7 +41,10 @@ class EnsureDataset(Constraint):
     the DataLad core package). With ``installed=False`` no exception is
     raised and a dataset instances matching PWD is returned.
     """
-    def __init__(self, installed: bool = None, purpose: str = None):
+    def __init__(self,
+                 installed: bool | None = None,
+                 purpose: str | None = None,
+                 idcheck: bool | None = None):
```

Comment:
What about `with_id: bool`? It would read "ensure dataset with id", not "ensure dataset idcheck". The code is als not checking the ID, only whether it is more than nothing.

Maybe "with" have too much of meaning in the Python context. `require_id` would also do it for me.

Reply:
I personally like ``require_id`` best

Reply:
Renamed it to ``require_id`` now.